### DISCUSSION TITLE
Allow enabling ViewTransitions components in configuration files

### DISCUSSION THREAD

**Discussion by LuHugo** [withastro/starlight#1032] [Feature Requests]
### What version of `starlight` are you using?

0.11.2

### What is your idea?

There seems to be no other way to enable the ViewTransitions component other than overriding the component.If a theme logo is set, the logo in the header will be displayed dynamically according to the theme, resulting in a large jitter when switching pages. You can enable ViewTransitions to solve this problem. But there seems to be no other way to add ViewTransitions other than overriding the component.

### Why is this feature necessary?

Add ViewTransitions to the page

### Participation

- [ ] I am willing to submit a pull request for this feature.

**Comment by delucis** [comment]
Thanks for the issue @LuHugo! https://github.com/withastro/starlight/pull/694 is trying to implement this currently. It’s not super simple as we would need to add some handling to ensure all the client-side scripts work as expected (as you observed with the theme jitter) and we need to do that in a way that works well also for people not using view transitions. But you can follow along with progress there!
